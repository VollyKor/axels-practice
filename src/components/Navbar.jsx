import { Container, Row, Col } from "react-bootstrap";

import { NavLink, Navbar, NavLogo, NavText } from "../styled/NavBar";
import { CartLink } from "components";

export default function NavBar() {
  return (
    <Container className="mb-5">
      <Row>
        <Col className="mx-auto">
          <Navbar className="justify-content-around">
            <NavLink to="/">
              <NavLogo className="mr-1">{"</>"}</NavLogo>
              <NavText className="h5">Front-end Developer Test Task</NavText>
            </NavLink>
            <NavLink to="/cart">
              <CartLink />
            </NavLink>
          </Navbar>
        </Col>
      </Row>
    </Container>
  );
}
