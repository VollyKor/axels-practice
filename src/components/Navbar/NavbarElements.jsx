import styled from "styled-components";
import { Nav } from "react-bootstrap";
import { LinkContainer } from "react-router-bootstrap";

function C_Navbar(props) {
  return <Nav {...props}>{props.children}</Nav>;
}

function C_NavBarLink({ children, to }) {
  return (
    <LinkContainer to={to} exact>
      <Nav.Link className="">{children}</Nav.Link>
    </LinkContainer>
  );
}

export const Navbar = styled(C_Navbar)`
  background: rgb(255, 255, 255);
  border-bottom: 1px solid #c26ea9;
`;

export const NavLink = styled(C_NavBarLink)`
  color: #c26ea9;
`;

export const NavLogo = styled.span`
  color: #fe71ac;
`;

export const NavText = styled(NavLogo)`
  color: #9362b9;
`;
