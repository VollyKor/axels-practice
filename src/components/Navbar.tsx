import { Container, Row, Col } from 'react-bootstrap';

import { CartLink } from 'components';
import { NavLink } from 'styled/NavBarLink';
import { Navbar, NavLogo, NavText } from 'styled/NavBar';

const NavBar = () => (
    <Container className="mb-5">
        <Row>
            <Col className="mx-auto">
                <Navbar>
                    <NavLink to="/">
                        <NavLogo className="mr-1">{'</>'}</NavLogo>
                        <NavText className="h5">
                            Front-end Developer Test Task
                        </NavText>
                    </NavLink>

                    <NavLink to="/cart/shipping">
                        <CartLink />
                    </NavLink>
                </Navbar>
            </Col>
        </Row>
    </Container>
);

export default NavBar;
