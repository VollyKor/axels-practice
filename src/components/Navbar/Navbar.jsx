import { Container, Row, Col } from "react-bootstrap";
import { NavLink, Navbar, NavLogo, NavText } from "./NavbarElements";
import Cart from "../CartLink/CartLink";

export default function NavBar() {
  return (
    <Container className="mb-5">
      <Row>
        <Col className="mx-auto">
          <Navbar className="justify-content-around">
            <NavLink to="/">
              <NavLogo className="mr-1">{"</>"}</NavLogo>
              <NavText className="h5">Front-end Developer Test Task</NavText>
            </NavLink>
            <NavLink to="/cart">
              <Cart />
            </NavLink>
          </Navbar>
        </Col>
      </Row>
    </Container>
  );
}
