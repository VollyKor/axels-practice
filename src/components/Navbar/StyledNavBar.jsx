import { Container } from "react-bootstrap";
import { Nav, Link, Icon, Cart } from "./NavbarElements";
import { IoCartOutline } from "react-icons/io5";

export default function StyledNavBar() {
  return (
    <header>
      <Container>
        <Nav>
          <Link to="/">
            <Icon />
            <span>{"</>"}</span>
            <span>Front-end Developer Test Task</span>
          </Link>
          <Link to="/cart">
            <span>Cart</span>
            <IoCartOutline size="25px" />
          </Link>
        </Nav>
      </Container>
    </header>
  );
}
