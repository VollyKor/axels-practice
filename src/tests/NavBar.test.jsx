import { shallow } from 'enzyme';

import Navbar from '../components/Navbar';

it('renders correctly', () => {
    const wrapper = shallow(<Navbar />);
    expect(wrapper).toMatchSnapshot();
});

// Work incorrect
test('testing route links', () => {
    const wrapper = shallow(<Navbar />);

    const Cart = wrapper.find('CartLink');
    expect(Cart.exists()).toBe(true);
});
