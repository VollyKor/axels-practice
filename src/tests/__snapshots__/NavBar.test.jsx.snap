// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Container
  className="mb-5"
  fluid={false}
>
  <Row
    noGutters={false}
  >
    <Col
      className="mx-auto"
    >
      <Styled(C_Navbar)>
        <Styled(C_NavBarLink)
          to="/"
        >
          <styled.span
            className="mr-1"
          >
            &lt;/&gt;
          </styled.span>
          <Styled(styled.span)
            className="h5"
          >
            Front-end Developer Test Task
          </Styled(styled.span)>
        </Styled(C_NavBarLink)>
        <Styled(C_NavBarLink)
          to="/cart/shipping"
        >
          <CartLink />
        </Styled(C_NavBarLink)>
      </Styled(C_Navbar)>
    </Col>
  </Row>
</Container>
`;
